bylaws/008-psr-evolution.md

```diff
@@ -0,0 +1,71 @@
+# PSR evolution
+
+## Scope and objectives
+
+A PSR is often comprised of text and code, more specifically interfaces. Those interfaces are pieces of code that are released and tagged at a specific moment in time, but the PHP language doesn't stand still... it evolves over time.
+
+This means that those interfaces need to withstand that evolution, and sometimes they need to be updated, to leverage new language features that could help better enforce the behaviors exposed in the PSR itself.
+
+At the same time, a PSR cannot be changed after its release (only erratas allowed), to protect a package that declared compatibility from becoming de-facto not compatible anymore.
+
+This document defines a process to be followed in updating PSR interfaces, in a way that is not breaking in regard to behavior for end users, and with an appropriate upgrade path for the packages.
+
+## New releases
+
+A new minor release of a PHP-FIG package containing interfaces for a PSR MUST follow these rules:
+ * the new release MUST follow [Semantic Versioning](https://semver.org/) rules;
+ * the PSR behavior MUST NOT be altered;
+ * the packages that implement the interfaces or users that are consuming them MUST NOT suffer breaking changes;
+ * the PHP version constraint of the PHP-FIG package MAY be altered to require newer language features that would aid cross-compatibility;
```

Comment:
I think you need to define what is meant by the term "cross-compatibility" before using it here. Are we talking "forwards compatibility", whereby the code adds features that would make it compatible with later versions of PHP, while retaining compatibility with current versions? Or are we talking compatibility with both consumer and implementor? or something else?

Reply:
I wrote this thinking about the current specific case of bumping to 7.2 to allow dropping types. Would a reword like this one work for you?

```suggestion
 * the PHP version constraint of the PHP-FIG package MAY be increased to leverage newer language features, especially those that would aid cross-compatibility of consumers with the old and the new versions;
```

Reply:
That sounds perfect!

Reply:
Only part is: I'd put that in a "Terminology" section at the beginning (we do that in a few PSRs already, so it will be familiar to PSR readers). That way, when it's referenced elsewhere, the context is fully known.

Reply:
I've added that with 94b5cc79e6b9c35788737dd76164f87690a0cc71, is it ok?

Reply:
@Jean85 Please either commit your suggestion (which looks good to me as well) or mark this conversation as resolved. The Terminology section helps, but it wouldn't hurt to have the extra clarity here.